grakn-graph/src/main/java/ai/grakn/graph/internal/Cache.java

```diff
@@ -110,10 +114,14 @@ void ifPresent(Consumer<V> modifier){
     }
 
     /**
-     * Takes the current value in the transaction cache if it is present and puts it in the sharedValue reference so
+     * Takes the current value in the transaction cache if it is present and puts it in the valueGlobal reference so
      * that it can be accessed via all transactions.
      */
     void flush(){
-        if(isPresent()) sharedValue = Optional.of(get());
+        if(isPresent()) {
+            V newValue = get();
+            if(!valueGlobal.isPresent() || !valueGlobal.get().equals(newValue)) valueGlobal = Optional.of(get());
```

Comment:
Got it!

But now I confused myself again. What's the check for? I'm starting to think it doesn't really do anything.